packages/bun-types/devserver.d.ts

```diff
@@ -0,0 +1,24 @@
+export {};
+
+declare global {
+  interface ImportMeta {
+    /**
+     * Hot module replacement
+     *
+     * https://bun.sh/docs/bundler/fullstack
+     */
+    hot: {
+      /**
+       * import.meta.hot.data maintains state between module instances during hot replacement, enabling data transfer from previous to new versions.
+       *
+       * @example
+       * ```ts
+       * import.meta.hot.data = {
+       *   bun: 'is cool',
+       * };
+       * ```
+       */
+      data: any;
```

Comment:
@paperclover said it was `Record<PropertyKey, any>`, but doesn't really matter that much as you can set it to whatever (its just default an object)